model_analyzer/record/metrics_manager.py

```diff
@@ -559,9 +559,20 @@ def _get_gpu_inference_metrics(self):
         # Stop and destroy DCGM monitor
         gpu_records = self._gpu_monitor.stop_recording_metrics()
 
+        if not gpu_records and self._expected_gpu_records():
+            raise TritonModelAnalyzerException(
+                f'No GPU metrics returned. Please check that the `triton_metrics_url` value is set correctly.'
+            )
+
         gpu_metrics = self._aggregate_gpu_records(gpu_records)
         return gpu_metrics
 
+    def _expected_gpu_records(self) -> bool:
```

Comment:
I think this is poorly named, or missing extra conditions. Based on its name, it should also look at 'cpu_only'. I know that flag is evaluated earlier (and that this function will only be called if cpu_only is false), but this could be confusing to a developer in the future.

Also, as a follow up story, remote should work fine. Not sure about c_api yet.

Also, as a follow up story (related to remote), we should consider changing the default metrics URL to be based off of the triton server url.  You can look at the recent PA change I made around this [here](https://github.com/triton-inference-server/client/pull/338/files).  For normal PA runs, MA wouldn't need to change anything. However, to get the server-only metrics it would need a similar change.